clash-lib/src/Clash/Normalize/Transformations/Inline.hs

```diff
@@ -379,6 +382,45 @@ inlineCleanup (TransformContext is0 _) (Letrec binds body) = do
 inlineCleanup _ e = return e
 {-# SCC inlineCleanup #-}
 
+-- | Takes a binding and collapses its term if it is a noop
+inlineIdentity :: HasCallStack => NormRewrite
+inlineIdentity _ (Letrec binds body) = do
+  (untouched,collapsed) <- partitionEithers <$> mapM identityAnalysis binds
+  if null collapsed
+    then return (Letrec binds body)
+    else changed (Letrec (untouched ++ collapsed) body)
```

Comment:
Perhaps we can leave the order of the let-bindings untouched by changing `identityAnalysis` slightly:
```haskell
inlineIdentity _ (Letrec binds body) = do
  binds1 <- mapM identityAnalysis binds
  return (Letrec binds1 body)
```

Reply:
Intuitively I assumed that I needed to call `changed` in case identities have been found and collapsed.
So yeah, if that's not necessary, this can be simplified.